### ISSUE TITLE
In DSpace 8, no new item can be deposited

### ISSUE THREAD

**Issue by pilasou** [DSpace/dspace-angular#2920]
**Describe the bug**
I used the sandbox and submit with Demo Submitter account in the collection "New collection created as Site Administrator" (https://sandbox.dspace.org/handle/10673/1159). I add an item using the Pubmed Import feature of DSpace. I am able to save the item for later, but each time I try to deposit the item, I get an error:   "There was an error when submitting the item, please try again." 

Can save/save for later and find the submission in "My DSPace."

**To Reproduce**
1. As submitter, submit a new item in a collection (Entity type Publication)
2. Fullfill all mandatory metadata, add a file and aproove licence
3. Clic on Deposit, 
4. The error displays on screen and the item cannot be deposited.

**Comment by tdonohue** [comment]
I've dug into this. 
It appears to be an issue with sending emails from DSpace 8 after the backend was upgraded to Jakarta EE in https://github.com/DSpace/DSpace/pull/9321.  The error occurs in *sending* an email notification, and it appears to block the submission from completing.  Here's the error in the dspace.log:
```
2024-04-11 12:25:54,727 ERROR 541bfbc1-43a4-425c-9eda-66f76bfa7d05 b13c8585-952a-4083-8978-9f9e03b241e2 org.dspace.app.rest.exception.DSpaceApiExceptionControllerAdvice @ An exception has occurred (status:500)
 jakarta.servlet.ServletException: Handler dispatch failed: java.lang.IllegalAccessError: failed to access class com.sun.activation.registries.LogSupport from class jakarta.activation.MailcapCommandMap (com.sun.activation.registries.LogSupport and jakarta.activation.MailcapCommandMap are in unnamed module of loader org.apache.catalina.loader.ParallelWebappClassLoader @5b6ec132)
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1104) ~[spring-webmvc-6.1.5.jar:6.1.5]
...
Caused by: java.lang.IllegalAccessError: failed to access class com.sun.activation.registries.LogSupport from class jakarta.activation.MailcapCommandMap (com.sun.activation.registries.LogSupport and jakarta.activation.MailcapCommandMap are in unnamed module of loader org.apache.catalina.loader.ParallelWebappClassLoader @5b6ec132)
        at jakarta.activation.MailcapCommandMap.<init>(MailcapCommandMap.java:149) ~[jakarta.activation-2.0.1.jar:2.0.1]
        at jakarta.activation.CommandMap.getDefaultCommandMap(CommandMap.java:55) ~[jakarta.activation-2.0.1.jar:2.0.1]
        at jakarta.activation.DataHandler.getCommandMap(DataHandler.java:127) ~[jakarta.activation-2.0.1.jar:2.0.1]
        at jakarta.activation.DataHandler.getDataContentHandler(DataHandler.java:589) ~[jakarta.activation-2.0.1.jar:2.0.1]
        at jakarta.activation.DataHandler.writeTo(DataHandler.java:289) ~[jakarta.activation-2.0.1.jar:2.0.1]
        at jakarta.mail.internet.MimeUtility.getEncoding(MimeUtility.java:316) ~[jakarta.mail-2.0.1.jar:2.0.1]
        at jakarta.mail.internet.MimeBodyPart.updateHeaders(MimeBodyPart.java:1551) ~[jakarta.mail-2.0.1.jar:2.0.1]
        at jakarta.mail.internet.MimeMessage.updateHeaders(MimeMessage.java:2246) ~[jakarta.mail-2.0.1.jar:2.0.1]
        at jakarta.mail.internet.MimeMessage.saveChanges(MimeMessage.java:2206) ~[jakarta.mail-2.0.1.jar:2.0.1]
        at jakarta.mail.Transport.send(Transport.java:99) ~[jakarta.mail-2.0.1.jar:2.0.1]
        at org.dspace.core.Email.send(Email.java:482) ~[dspace-api-8.0-SNAPSHOT.jar:8.0-SNAPSHOT]
```

This was likely not found previously because many developers will *disable* email sending on local dev machines.  I'm looking into a possible fix.  It's likely a minor dependency error.

**Comment by mwoodiupui** [comment]
In lib/ there's a mess of different Java Activation JARs:  angus-activation (the new new one) and jakarta.activation (the old new one) as well as javax.activation (the old one).  Looks like Jakarta Mail pulls in jakarta.activation and jaxb-runtime pulls in angus-activation.  Maybe we should be using org.eclipse.angus:jakarta.mail rather than com.sun.mail:jakarta.mail?

**Comment by tdonohue** [comment]
@mwoodiupui :
> Maybe we should be using org.eclipse.angus:jakarta.mail rather than com.sun.mail:jakarta.mail?

This is exactly the bug fix I'm working on.  It's a dependency issue which I failed to discover as I have email turned off locally.  I think I nearly have a PR ready & it does move us over to `org.eclipse.angus:jakarta.mail`....working on the final testing

**Comment by pilasou** [comment]
Hi @tdonohue, thanks! Has the fix been deployed on the sandox?

**Comment by tdonohue** [comment]
@pilasou : Yes, this fix was deployed to sandbox...however, it seems it was only a *partial fix* and the Sandbox email setup requires more.  So, while I did get rid of the [above error](https://github.com/DSpace/dspace-angular/issues/2920#issuecomment-2050088989), a *new error* is now appearing:
```
2024-04-12 09:55:48,819 ERROR 541bfbc1-43a4-425c-9eda-66f76bfa7d05 9695e190-55ec-45e4-aede-d6188b01e8e4 org.dspace.app.rest.RestResourceController @ Something went wrong whilst creating the object for apiCategory: workflow and model: workflowitems
 java.lang.ClassCastException: class org.apache.geronimo.mail.handlers.TextHandler cannot be cast to class jakarta.activation.DataContentHandler (org.apache.geronimo.mail.handlers.TextHandler and jakarta.activation.DataContentHandler are in unnamed module of loader org.apache.catalina.loader.ParallelWebappClassLoader @413f69cc)
        at jakarta.activation.MailcapCommandMap.getDataContentHandler(MailcapCommandMap.java:620) ~[jakarta.activation-api-2.1.3.jar:?]
        at jakarta.activation.MailcapCommandMap.createDataContentHandler(MailcapCommandMap.java:573) ~[jakarta.activation-api-2.1.3.jar:?]
        at jakarta.activation.DataHandler.getDataContentHandler(DataHandler.java:591) ~[jakarta.activation-api-2.1.3.jar:?]
        at jakarta.activation.DataHandler.writeTo(DataHandler.java:290) ~[jakarta.activation-api-2.1.3.jar:?]
        at jakarta.mail.internet.MimeUtility.getEncoding(MimeUtility.java:318) ~[jakarta.mail-2.0.3.jar:?]
        at jakarta.mail.internet.MimeBodyPart.updateHeaders(MimeBodyPart.java:1562) ~[jakarta.mail-2.0.3.jar:?]
        at jakarta.mail.internet.MimeMessage.updateHeaders(MimeMessage.java:2272) ~[jakarta.mail-2.0.3.jar:?]
        at jakarta.mail.internet.MimeMessage.saveChanges(MimeMessage.java:2232) ~[jakarta.mail-2.0.3.jar:?]
        at jakarta.mail.Transport.send(Transport.java:101) ~[jakarta.mail-2.0.3.jar:?]
        at org.dspace.core.Email.send(Email.java:482) ~[dspace-api-8.0-SNAPSHOT.jar:8.0-SNAPSHOT]
        at org.dspace.xmlworkflow.XmlWorkflowServiceImpl.notifyOfArchive(XmlWorkflowServiceImpl.java:682) ~[dspace-api-8.0-SNAPSHOT.jar:8.0-SNAPSHOT]
        at org.dspace.xmlworkflow.XmlWorkflowServiceImpl.archive(XmlWorkflowServiceImpl.java:627) ~[dspace-api-8.0-SNAPSHOT.jar:8.0-SNAPSHOT]
        at org.dspace.xmlworkflow.XmlWorkflowServiceImpl.start(XmlWorkflowServiceImpl.java:226) ~[dspace-api-8.0-SNAPSHOT.jar:8.0-SNAPSHOT]
        at org.dspace.xmlworkflow.XmlWorkflowServiceImpl.start(XmlWorkflowServiceImpl.java:88) ~[dspace-api-8.0-SNAPSHOT.jar:8.0-SNAPSHOT]
```

So, it seems this requires an additional fix to get the deposit process fully working.    Locally, the deposit process works perfectly if email notifications are turned off (`mail.server.disabled=true`).  But, turning on the email notifications (which obviously we need) has resulted in these unexpected errors.  I'll reopen this ticket and look to fix it in a follow-up PR.

**Comment by mwoodiupui** [comment]
abdera-core and axiom-api are pulling in rather old Geronimo bits.  I thought we got rid of abdera?

**Comment by tdonohue** [comment]
@mwoodiupui : Just to answer your question for the record. Abdera is now pulled back in by `io.gdcc:sword2-server` new SWORDv2 library.  That library is an updated version of the (really old) SWORDv2 Java API which is now updated to Jakarta EE.  It is possible we'd want to find a way to re-remove Abdera, but I've found SWORDv2 challenging in general...it's API is unmaintained and **relies heavily on Abdera API**.  The only way to remove Abdera entirely is essentially to build our own custom SWORDv2 Java API (which it may come to), and there's no clear replacement for Abdera anywhere out there.  For all the ugly backstory, see https://github.com/DSpace/DSpace/issues/8188.

Nonetheless, I've excluded Geronimo from being pulled in via that SWORDv2 dependency here: https://github.com/DSpace/DSpace/pull/9472  I'm hoping this will finally solve the email issues 🤞
